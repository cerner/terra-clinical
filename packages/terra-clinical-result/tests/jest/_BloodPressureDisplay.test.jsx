import React from 'react';
import BloodPressureDisplay from '../../src/_BloodPressureDisplay';
import { DefaultBloodPressureResult } from '../../src/terra-dev-site/test/clinical-result/TestResults';

describe('BloodPressureDisplay', () => {
  const defaultResult = {
    ...DefaultBloodPressureResult.systolic,
    statusInError: false,
    cleanedUnit: 'mmhg',
    noData: false,
  };
  const {
    id,
  } = DefaultBloodPressureResult;
  const type = 'Systolic';

  it('should render a default BloodPressureDisplay', () => {
    const display = shallow(<BloodPressureDisplay result={defaultResult} id={id} type={type} />);
    expect(display).toMatchSnapshot();
  });

  it('should render a ResultError', () => {
    const display = shallow(<BloodPressureDisplay id={id} type={type} />);
    expect(display).toMatchSnapshot();
  });

  it('should render a NoData', () => {
    const result = {
      ...defaultResult,
      noData: true,
    };
    const display = shallow(<BloodPressureDisplay result={result} id={id} type={type} />);
    expect(display).toMatchSnapshot();
  });

  it('shouldn\'t render interpretation if statusInError', () => {
    const result = {
      ...defaultResult,
      statusInError: true,
      interpretation: 'critical',
    };
    const display = shallow(<BloodPressureDisplay result={result} id={id} type={type} />);
    expect(display).toMatchSnapshot();
  });

  it('should hide the unit if hideUnit is true', () => {
    const display = shallow(<BloodPressureDisplay result={defaultResult} id={id} type={type} hideUnit />);
    expect(display).toMatchSnapshot();
  });

  it('should hide the unit if data\'s cleanedUnit is equal to diastolicUnit prop', () => {
    const display = shallow(<BloodPressureDisplay result={defaultResult} id={id} type={type} diastolicUnit={defaultResult.cleanedUnit} />);
    expect(display).toMatchSnapshot();
  });

  it('shouldn\'t hide the unit if data\'s statusInError is true', () => {
    const result = {
      ...defaultResult,
      statusInError: true,
      interpretation: 'critical',
    };
    const display = shallow(<BloodPressureDisplay result={result} id={id} type={type} diastolicUnit={defaultResult.cleanedUnit} />);
    expect(display).toMatchSnapshot();
  });
});
