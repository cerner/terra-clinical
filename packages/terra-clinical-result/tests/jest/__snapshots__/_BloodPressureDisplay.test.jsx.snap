// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BloodPressureDisplay should hide the unit if data's cleanedUnit is equal to diastolicUnit prop 1`] = `
<ConditionalWrapper
  condition={false}
  key="del-Systolic-111.1"
  wrapper={[Function]}
>
  <Observation
    eventId="111.1"
    hideUnit={true}
    key="Observation-Systolic-111.1"
    result={
      Object {
        "unit": "mmHg",
        "value": "180",
      }
    }
  />
</ConditionalWrapper>
`;

exports[`BloodPressureDisplay should hide the unit if hideUnit is true 1`] = `
<ConditionalWrapper
  condition={false}
  key="del-Systolic-111.1"
  wrapper={[Function]}
>
  <Observation
    eventId="111.1"
    hideUnit={true}
    key="Observation-Systolic-111.1"
    result={
      Object {
        "unit": "mmHg",
        "value": "180",
      }
    }
  />
</ConditionalWrapper>
`;

exports[`BloodPressureDisplay should render a NoData 1`] = `
<InjectIntl(KnownNoData)
  key="NoData-Systolic-111"
/>
`;

exports[`BloodPressureDisplay should render a ResultError 1`] = `
<InjectIntl(ResultError)
  key="Error-Systolic-111"
/>
`;

exports[`BloodPressureDisplay should render a default BloodPressureDisplay 1`] = `
<ConditionalWrapper
  condition={false}
  key="del-Systolic-111.1"
  wrapper={[Function]}
>
  <Observation
    eventId="111.1"
    hideUnit={false}
    key="Observation-Systolic-111.1"
    result={
      Object {
        "unit": "mmHg",
        "value": "180",
      }
    }
  />
</ConditionalWrapper>
`;

exports[`BloodPressureDisplay shouldn't hide the unit if data's statusInError is true 1`] = `
<ConditionalWrapper
  condition={true}
  key="del-Systolic-111.1"
  wrapper={[Function]}
>
  <Observation
    eventId="111.1"
    hideUnit={false}
    interpretation={null}
    key="Observation-Systolic-111.1"
    result={
      Object {
        "unit": "mmHg",
        "value": "180",
      }
    }
  />
</ConditionalWrapper>
`;

exports[`BloodPressureDisplay shouldn't render interpretation if statusInError 1`] = `
<ConditionalWrapper
  condition={true}
  key="del-Systolic-111.1"
  wrapper={[Function]}
>
  <Observation
    eventId="111.1"
    hideUnit={false}
    interpretation={null}
    key="Observation-Systolic-111.1"
    result={
      Object {
        "unit": "mmHg",
        "value": "180",
      }
    }
  />
</ConditionalWrapper>
`;
